pkg/reconciler/handlers.go

```diff
@@ -12,3 +14,33 @@ func Handler(h func(interface{})) cache.ResourceEventHandler {
 		DeleteFunc: h,
 	}
 }
+
+type ErrorHandler interface {
+	OnReconcileError(err error) error
+}
+
+func NewErrorHandler(retryAfter time.Duration) ErrorHandler {
+	return &retryErrorHandler{
+		retryDuration: retryAfter,
+	}
+}
+
+type retryErrorHandler struct {
+	retryDuration time.Duration
+}
+
+func (r *retryErrorHandler) OnReconcileError(err error) error {
+	var networkError *NetworkError
+	if errors.As(err, &networkError) {
+		return controller.NewRequeueAfter(r.retryDuration)
```

Comment:
My understanding is that that underlying reconciler will automatically requeue errors with an exponential backoff. Do we need to explicitly provide the NewRequeueAfter with a duration?

Reply:
Won't an explicit duration also prevent the backoff?

Reply:
I think you are right, I was checking the following [code](https://github.com/knative/pkg/blob/main/controller/controller.go#L560-L576) in the knative package. I misunderstood creating a *requeueKeyError* was mandatory. 
If just re-throw the transient error it actually be re-queued